Formula/unionfs-fuse.rb

```diff
@@ -0,0 +1,22 @@
+class UnionfsFuse < Formula
+  desc "FUSE based union filesystem with COW"
+  homepage "https://github.com/rpodgorny/unionfs-fuse"
+  url "https://github.com/rpodgorny/unionfs-fuse/archive/v2.1.tar.gz"
+  sha256 "c705072a33a18cbc73ffe799331d43410b6deef5d6f2042038f8fd3ab17b6e2e"
+  head "https://github.com/rpodgorny/unionfs-fuse.git"
+
+  depends_on "pkg-config" => :build
+  depends_on :osxfuse
+
+  def install
+    system "make"
+    bin.install "src/unionfs"
+    bin.install "src/unionfsctl"
+    sbin.install "mount.unionfs"
+    man8.install "man/unionfs.8"
+  end
+
+  test do
+    system "#{bin}/unionfs", "-V"
```

Comment:
What does this do?

Reply:
Prints the version info for the installed binary. It checks that it is compiled correctly and that FUSE is available as it queries FUSE's version and should fail if it is not available.

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.